app/main.py

```diff
@@ -1,2 +1,100 @@
+        if self.length <= round(self.hash_capacity * 2 / 3):
+            index = hash(key) % self.hash_capacity
+            if self.hash_table[index] is None:
+                self.hash_table[index] = (key, value, hash(key))
+                self.length += 1
+            elif self._check_key(key, index):
+                self.hash_table[index] = (key, value, hash(key))
+            else:
+                while (self.hash_table[index] is not None
+                       and not self._check_key(key, index)):
+                    index = (index + 1) % self.hash_capacity
+                if self.hash_table[index] is None:
+                    self.hash_table[index] = (key, value, hash(key))
+                    self.length += 1
+                else:
+                    self.hash_table[index] = (key, value, hash(key))
+        else:
+            self.resize()
+            self[key] = value
+
+    def resize(self) -> None:
+        self.hash_capacity *= 2
+        hash_table = self.hash_table.copy()
+        self.hash_table = [None] * self.hash_capacity
+        self.length = 0
+        for place in hash_table:
+            if place:
+                self[place[0]] = place[1]
+
+    def __getitem__(self, key: hashable) -> Any:
+        index = hash(key) % self.hash_capacity
+        if self.hash_table[index] is None:
+            raise KeyError(f"There is no element with this key: {key}")
+        elif self._check_key(key, index):
+            return self.hash_table[index][1]
+        else:
+            while self.hash_table[index] and not self._check_key(key, index):
+                index = (index + 1) % self.hash_capacity
+            if self.hash_table[index] or self._check_key(key, index):
+                return self.hash_table[index][1]
+            raise KeyError(f"There is no element with this key: {key}")
+
+    def _check_key(self, key: hashable, index: int) -> bool:
+        return (hash(key) == self.hash_table[index][2]
+                and self.hash_table[index][0] == key)
+
+    def clear(self) -> None:
+        self.hash_table = [None] * 8
+        self.length = 0
+
+    def __delitem__(self, key: hashable) -> None:
+        if self[key]:
+            index = hash(key) % self.hash_capacity
+            if self._check_key(key, index):
+                self.hash_table[index] = None
+            else:
+                for i in range(len(self.hash_table)):
+                    if self.hash_table[i] and self._check_key(key, i):
+                        self.hash_table[i] = None
+        else:
+            raise KeyError(f"Wrong key: {key}")
+
+    def get(self, key: hashable) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return
+
+    def pop(self, key: hashable, *args: str) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            if not args:
+                raise
+            else:
+                default, = args
+                return default
```

Comment:
try to avoid using else here, also are u sure about *args as arg ? maybe this is should only one param ?
![image](https://user-images.githubusercontent.com/103201010/221874878-ba934b94-008f-4fcc-8467-1ce19171a3ad.png)